docs: update CONTRIBUTING.md and general documentation (#1098)
nvim-neorg/neorg: docs/CONTRIBUTING.md

### Coding style
- I use snake_case for everything, and you should too :P

> - **Please** comment all your code! Annotations for functions are generated by [neogen](https://github.com/danymat/neogen).

Reviewer: ```suggestion
- **Please** comment all your code! Annotations for functions can be generated with [neogen](https://github.com/danymat/neogen).
```

## Adding functionality
Whenever you are planning on extending neorg, try your best to add *all* extra functionality through modules and modules only. Make changes to the Neorg core only if absolutely necessary.
When adding stuff, use lua only. `vim.cmd` in extreme cases :)

## Helping with documentation

> Like mentioned above, the documentation (wiki) is generated via [neogen](https://github.com/danymat/neogen) using comments in the code.

Reviewer: neogen is just used for annotations of functions

Author: OK, I read through neogen README and understand neogen is just for function annotations

I guess my confusion came from this line
https://github.com/nvim-neorg/neorg/blob/main/docs/CONTRIBUTING.md?plain=1#L39
```
- **Please** comment all your code! Documentation for functions is generated by [neogen](https://github.com/danymat/neogen).
```

When I first read it, I thought the documentation meant the auto-generated wiki
But it actually meant annotations/explanations of the code (which is then
auto-generated into wiki pages)

should we change it as such?

```diff
- **Please** comment all your code! *Documentation* for functions is generated by [neogen](https://github.com/danymat/neogen).
+ **Please** comment all your code! *Annotations* for functions is generated by [neogen](https://github.com/danymat/neogen).
```

Reviewer: yes ig you could do that
even though it would be clear if you look at neogen ig

Reviewer 2: ```diff
- **Please** comment all your code! *Documentation* for functions is generated by [neogen](https://github.com/danymat/neogen).
+ **Please** comment all your code! *Annotations* for functions are generated by [neogen](https://github.com/danymat/neogen).
```

Make sure to switch up the `is` to `are` and yup this seems like a good change to me!

Author: ```suggestion
The documentations (wiki pages) are generated based on the comments in the code.
And all the settings in generating the wiki are in `docgen/`

To generate documentations locally, run `make local-documentation`. See
`Makefile` for more details If there's no error, you should then see a `wiki/`
folder.

```

With neogen confusion cleared out, how about this?

Author: > yes ig you could do that even though it would be clear if you look at neogen ig

It's best to assume the newcomer reading the docs could misunderstand even after going through neogen README just like I did.
So having the documentation as clear as possible would be the best for everyone :)

Reviewer 3: > It's best to assume the newcomer reading the docs could misunderstand even after going through neogen README just like I did.

Do not hesitate to tell me if neogen README is not clear enough, and what do you think could be added/changed in order to improve it :)

Author: > Do not hesitate to tell me if neogen README is not clear enough, and what do you think could be added/changed in order to improve it :)

It was the Neorg CONTRIBUTING.md that confused me, the Neogen README is fine for the most part
but sure! I will let you know if there's anything I think that can be improved

When adding stuff, use lua only. `vim.cmd` in extreme cases :)

## Helping with documentation

> The documentations (wiki pages) are generated based on the comments in the code. And all the settings in generating the wiki are in [`docgen/`](../docgen/).

Reviewer: ```suggestion
The documentations (wiki pages) are generated based on the comments in the code. All the logic for generating the wiki is in [`docgen/`](../docgen/).
```

> To generate documentations locally, run `make local-documentation` (see `Makefile` for more details). If there's no error, you should then see a `wiki/` folder generated.

Reviewer: ```suggestion
To generate documentations locally, run `make local-documentation` (see [`Makefile`](../Makefile) for more details). If there's no error, you should then see a `wiki/` folder generated.
```

> If you feel like improving the documentation, please make changes to the corresponding comments in the correct file. For example, to make changes to the `core.qol.todo_items` documentation, please make changes to the comments in [`./lua/neorg/modules/core/qol/todo_items/module.lua`](../lua/neorg/modules/core/qol/todo_items/module.lua)

Reviewer: ```suggestion
If you want to work on the documentation, please make changes to the corresponding comments in the correct file. For example, to make changes to the `core.qol.todo_items` documentation, please make changes to the comments in [`./lua/neorg/modules/core/qol/todo_items/module.lua`](../lua/neorg/modules/core/qol/todo_items/module.lua)
```
